client/auth/action-creators.ts

```diff
@@ -117,7 +97,7 @@ export function logIn(
     })
 
     if (result.user.locale && result.user.locale !== locale) {
```

Comment:
Hmm, not sure I understand. Do you mean a part of the `@auth/loadCurrentSession` action? Because I'm just calling the `i18n`'s `changeLanguage` here (which will internally try to best match the user's locale to one of our `TranslationLanguage`, using the fallback language if none is found), there's no reducer changes.

Reply:
No I mean in `changeLanguageLocally`. You're dispatching it, it's an action creator, whether or not it makes any reducer changes is irrelevant to that.

I want you to not perform this repeated condition around dispatching this in every place you call it, and instead just perform that check in the action creator. If we needed to add a 4th place that calls this, whoever adds that would *also* need to add this same condition (or potentially introduce weird bugs/behaviors in that place that don't happen in the others), which isn't good. It's better to just always call the function and have it no-op itself if nothing needs to be done (this is why I tend to create `maybeDoX` functions)